## Project Results

To find information about a project, go to its home page. Some information is available for all project types. This includes:

* **About** - The project type \(Mixed, Genomes, Metagenomes, Clade\) and which identity engines were used in creating the project
* **Project Progress** - The number of genomes for which processing is complete and incomplete
* **hAAI distance**s - heuristic estimates of average amino acid distances among genomes
* **AAI distances** - amino acid distances among genomes  

Clicking on the headings for the last two items presents the opportunity to download summary files of various types and perhaps displays a graph of the distribution of values. Click on **Learn more** for information about the file types.

For genome and clade projects, MiGA may also report the following, depending on results:

* **ANI distances** - average nucleotide identity between genomes with AAI â‰¥ 90%
* **Clade finding** - groups of genomes are found using AAI and ANI values. Click on **Learn more** for detailed information about how this is done and the types of files that can be downloaded.

For clade projects only, MiGA will report the following:

* **Sub-clades** - clades below the species level using ANI values. Click on the blue text following "MiGA proposes" for lists of genomes in each sub-clade. Click on **Learn more** for information about the various files that can be downloaded, including trees in Newick format. 
* **OGS** - groups of othology using reciprocal best matches between all pairs of datasets. Click on **Learn more** for information about the types of files that can be downloaded.

For indexed genome projects, **Project stats** displays a taxonomic tree of the genomes in the project. The reference databases available for download from the **Admin console** page are indexed genome projects.
